tee-worker/omni-executor/rpc-server/src/methods/omni/get_approve_agent_wallet_data.rs

```diff
@@ -0,0 +1,337 @@
+	pub user_auth: UserAuth,
+	pub client_id: String,
+	pub agent_address: String,
+	pub agent_name: Option<String>,
+	pub chain_id: ChainId,
+	pub is_testnet: bool,
+}
+
+#[derive(Serialize, Clone)]
+pub struct GetApproveAgentWalletDataResponse {
+	pub main_address: String,
+	pub approve_agent_data: ApproveAgentData,
+}
+
+#[derive(Serialize, Clone)]
+pub struct ApproveAgentData {
+	pub action: ApproveAgentAction,
+	pub nonce: u64,
+	pub signature: String,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct ApproveAgentAction {
+	#[serde(serialize_with = "serialize_hex")]
+	pub signature_chain_id: u64,
+	pub hyperliquid_chain: String,
+	pub agent_address: Address,
+	pub agent_name: Option<String>,
+	pub nonce: u64,
+}
+
+impl ApproveAgentAction {
+	fn domain(&self) -> Eip712Domain {
+		eip712_domain! {
+			name: "HyperliquidSignTransaction",
+			version: "1",
+			chain_id: self.signature_chain_id,
+			verifying_contract: Address::ZERO,
+		}
+	}
+
+	fn struct_hash(&self) -> B256 {
+		let items = (
+            keccak256("HyperliquidTransaction:ApproveAgent(string hyperliquidChain,address agentAddress,string agentName,uint64 nonce)"),
+            keccak256(&self.hyperliquid_chain),
+            &self.agent_address,
+            keccak256(self.agent_name.as_deref().unwrap_or("")),
+            &self.nonce
+        );
+		keccak256(items.abi_encode())
+	}
+
+	fn eip712_signing_hash(&self) -> B256 {
+		let mut digest_input = [0u8; 2 + 32 + 32];
+		digest_input[0] = 0x19;
+		digest_input[1] = 0x01;
+		digest_input[2..34].copy_from_slice(&self.domain().hash_struct()[..]);
+		digest_input[34..66].copy_from_slice(&self.struct_hash()[..]);
+		keccak256(digest_input)
+	}
+}
+
+pub fn register_get_approve_agent_wallet_data(module: &mut RpcModule<RpcContext>) {
+	module
+		.register_async_method("omni_getApproveAgentWalletData", |params, ctx, _| async move {
+			let params = params.parse::<GetApproveAgentWalletDataParams>().map_err(|e| {
+				error!("Failed to parse params: {:?}", e);
+				ErrorCode::ParseError
+			})?;
+
+			let (email, verification_code) = match &params.user_auth {
+				UserAuth::Email(code) => {
+					let UserId::Email(email) = &params.user_id else {
+						error!("User ID must be an email for Email authentication");
+						return Err(ErrorObject::from(ErrorCode::ParseError));
+					};
+					(email.clone(), code.clone())
+				},
+				_ => {
+					error!("Only email authentication is supported for this method");
+					return Err(ErrorObject::from(ErrorCode::InvalidParams));
+				},
+			};
+
+			verify_email_authentication(ctx.clone(), &params.client_id, &email, &verification_code)
+				.map_err(|_| {
+					error!("Failed to verify email authentication");
+					ErrorObject::from(ErrorCode::ServerError(AUTH_VERIFICATION_FAILED_CODE))
+				})?;
+
+			let identity = Identity::from_web2_account(&email, Web2IdentityType::Email);
+			let omni_account = identity.to_omni_account(&params.client_id);
```

Comment:
I recommend we use `UserId` -> `Identity` conversion directly: it's more actively maintained.